lib/camunda-platform/features/modeling/behavior/UpdateInputOutputBehavior.js

```diff
@@ -0,0 +1,89 @@
+import inherits from 'inherits';
+
+import {
+  getBusinessObject,
+  is
+} from 'bpmn-js/lib/util/ModelUtil';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+/**
+ * Camunda BPMN specific `camunda:inputOutput` behavior.
+ */
+export default function UpdateInputOutputBehavior(eventBus) {
+
+  CommandInterceptor.call(this, eventBus);
+
+  this.executed([ 'properties-panel.update-businessobject-list' ], function(context) {
+    const {
+      element
+    } = context;
+
+    const businessObject = getBusinessObject(element);
+    const inputOutput = getInputOutput(businessObject);
+    const extensionElements = businessObject.get('extensionElements');
+
+    // Remove camunda:inputOutput if there are no input/output parameters anymore.
+    if (inputOutput && isEmpty(inputOutput)) {
+      context.withRemovedInputOutput = extensionElements.values;
+
+      extensionElements.values = extensionElements.values.filter(function(element) {
```

Comment:
Can we please use a command here? If I want to intercept changes to `extensionElement` the current implementation gives me no chance to do that. I know it's implemented the same way in other behaviors, I'd like to see that fixed in the future too.

Reply:
I updated the behavior to work with a command as proposed. Also, I made sure we also cover other commands.